packages/router/src/grpc/view-protocol-server/delegations-by-address-index.test.ts

```diff
@@ -0,0 +1,347 @@
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { delegationsByAddressIndex } from './delegations-by-address-index';
+import { ViewService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/view/v1/view_connect';
+import {
+  HandlerContext,
+  createHandlerContext,
+  createContextValues,
+  PromiseClient,
+} from '@connectrpc/connect';
+import { stakingClientCtx } from '../../ctx';
+import { QueryService as StakingService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/core/component/stake/v1/stake_connect';
+import {
+  AssetMetadataByIdResponse,
+  BalancesResponse,
+  DelegationsByAddressIndexRequest,
+  DelegationsByAddressIndexRequest_Filter,
+  DelegationsByAddressIndexResponse,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+import { STAKING_TOKEN_METADATA } from '@penumbra-zone/constants';
+import {
+  ValidatorInfoRequest,
+  ValidatorInfoResponse,
+  ValidatorState_ValidatorStateEnum,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/stake/v1/stake_pb';
+import { asIdentityKey, getAmount, getValidatorInfoFromValueView } from '@penumbra-zone/getters';
+import { PartialMessage } from '@bufbuild/protobuf';
+import {
+  Metadata,
+  ValueView,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
+
+const mockBalances = vi.hoisted(() => vi.fn());
+vi.mock('./balances', () => ({
+  balances: mockBalances,
+}));
+
+vi.mock('./asset-metadata-by-id', () => ({
+  assetMetadataById: () =>
+    Promise.resolve(new AssetMetadataByIdResponse({ denomMetadata: new Metadata() })),
+}));
+
+const activeValidatorBech32IdentityKey =
```

Comment:
These huge setup sections of our test suites are killing me. I gotta figure out an elegant solution to this.